cpp/src/arrow/CMakeLists.txt

```diff
@@ -833,7 +830,11 @@ if(ARROW_FILESYSTEM)
                                 PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON)
   endif()
   if(ARROW_HDFS)
-    list(APPEND ARROW_FILESYSTEM_SRCS filesystem/hdfs.cc)
+    list(APPEND
+         ARROW_FILESYSTEM_SRCS
+         filesystem/hdfs.cc
+         filesystem/hdfs_io.cc
+         filesystem/hdfs_internal.cc)
```

Comment:
Don't we need to link to `arrow::hadoop` as was done above? cc @kou for advice

Reply:
Hm, yeah. I will add a link as above as it makes sense.

Reply:
Ah, it is there already (that explains why nothing failed =) )
https://github.com/apache/arrow/blob/53ef438333efd4a6eb1421b3e812d10e5a8ed633/cpp/src/arrow/CMakeLists.txt#L857-L861

Not sure if the line with `CMAKE_DL_LIBS` is also needed here then?